raycicmd/dep_graph.go

```diff
@@ -0,0 +1,43 @@
+package raycicmd
+
+type depNode interface {
+	deps() []string
+	mark() // mark if this node should be included
+}
+
+type depGraph interface {
+	depNode(id string) depNode
+}
+
+func markDeps(g depGraph, set map[string]struct{}) {
+	hit := make(map[string]struct{})
+	for id := range set {
+		hit[id] = struct{}{}
```

Comment:
since the value of the map is not used, can we use type boolean or int (instead of struct{}{})

Reply:
changed to bool values.

technically `struct{}` takes 0 space, where bool takes 1 byte. but does not matter for this case.